app/models/chat_channel.rb

```diff
@@ -32,6 +32,21 @@ class ChatChannel < ActiveRecord::Base
         }
 
   class << self
+    def type_for(chatable)
+      case chatable
+      when Category
+        CategoryChannel
+      when DirectMessageChannel
+        DMChannel
+      else
+        raise "Unknown chatable type #{chatable.class}"
+      end
+    end
+
+    def for(chatable)
+      type_for(chatable).new(chatable: chatable)
+    end
+
     def public_channel_chatable_types
```

Comment:
Will we remove/refactor this at some point?

Reply:
We are clearly not yet on the final target, trying to make this in multiple PRs, chatable is still very much present, but yes we will most likely refactor this at some point too :+1: